### PR TITLE
Issue #3070 - Add additional logging of experiment branch and IP

### PR DESCRIPTION
**Pull request by miketaylr**
Tested locally:

```
[2019-11-12 14:35:18,817] INFO in views: 127.0.0.1 b'http://example.com'
[2019-11-12 14:35:18,818] INFO in helpers: is_valid_issue_form: form[submit_type] => empty submit_type value
[2019-11-12 14:35:18,818] INFO in helpers: is_valid_issue_form: missing param(s)? => set()
[2019-11-12 14:35:18,818] INFO in helpers: is_valid_issue_form: experiment branch => form-v1
[2019-11-12 14:35:18,818] INFO in views: 400: POST request w/o valid form (is_valid_issue_form).
```

### CODE REVIEW

**Comment by miketaylr** [comment]
oops tests

File: webcompat/helpers.py [hunk 1]
```diff
@@ -592,9 +592,12 @@ def is_valid_issue_form(form):
     valid_form = parameters_check and values_check
     if not valid_form:
         log.info('is_valid_issue_form: form[submit_type] => {0}'.format(
-            form.get('submit_type', 'missing submit_type value')))
+            form.get('submit_type') or 'empty submit_type value'))
```
Comment (karlcow) [inline webcompat/helpers.py]: Trying to understand which case you were trying to cover. 

```python
>>> form = {'foo': 'yeah', 'bar': 'boo'}
>>> form.get('foo', 'lol')
'yeah'
>>> form.get('foo') or 'lol'
'yeah'
>>> form = {'bar': 'boo'}
>>> form.get('foo', 'lol')
'lol'
>>> form.get('foo') or 'lol'
'lol'
>>> form = {}
>>> form.get('foo', 'lol')
'lol'
>>> form.get('foo') or 'lol'
'lol'
>>> form = {'foo': '', 'bar': 'boo'}
>>> form.get('foo', 'lol')
''
>>> form.get('foo') or 'lol'
'lol'
```

Comment (miketaylr) [reply webcompat/helpers.py]: basically this:

```
>>> f = {'a': ''}
>>> 'cool {0}'.format(f.get('a'))
'cool '
>>> 'cool {0}'.format(f.get('b'))
'cool None'
>>> 'cool {0}'.format(f.get('a') or 'dude')
'cool dude'
>>> 
```

**Comment by miketaylr** [comment]
> Let's try I would have probably added the IP for bot reasons.

Might as well do that now.

**Comment by karlcow** [comment]
thanks @miketaylr let's merge.

**Comment by miketaylr** [comment]
Thanks, I'll probably deploy tonight or tomorrow at some point.

**Comment by karlcow** [comment]
> Thanks, I'll probably deploy tonight or tomorrow at some point.

I would have done it, but given that I'm in and out. If anything goes wrong, I will not be able to intervene. Thanks mike.

**Comment by miketaylr** [comment]
Yeah, no worries -- I know you're travelling. And I wanted for the morning so we didn't have to sleep on any bad errors. :)

**Pull request merged**